source/Applications/Basic/Visualization/CaptureWritePCLVis3D/CaptureWritePCLVis3D.cpp

```diff
@@ -1,18 +1,121 @@
+        for(size_t i = 0; i < data.size(); ++i)
+        {
+            pointCloud->points[i].x = data(i).point.x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].y = data(i).point.y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].z = data(i).point.z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].r = data(i).color.r; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].g = data(i).color.g; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].b = data(i).color.b; // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
+        return pointCloud;
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGBNormal>::Ptr convertToPCLPointCloudWithNormals(
+        const Zivid::Array2D<Zivid::PointXYZColorRGBA> &data,
+        const Zivid::Array2D<Zivid::NormalXYZ> &normals)
+    {
+        auto pointCloudWithNormals = boost::make_shared<pcl::PointCloud<pcl::PointXYZRGBNormal>>();
+        pointCloudWithNormals->width = data.width();
+        pointCloudWithNormals->height = data.height();
+        pointCloudWithNormals->is_dense = false;
+        pointCloudWithNormals->points.resize(data.size());
+        for(size_t i = 0; i < data.size(); ++i)
+        {
+            pointCloudWithNormals->points[i].x = data(i).point.x;     // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudWithNormals->points[i].y = data(i).point.y;     // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudWithNormals->points[i].z = data(i).point.z;     // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudWithNormals->points[i].r = data(i).color.r;     // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudWithNormals->points[i].g = data(i).color.g;     // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudWithNormals->points[i].b = data(i).color.b;     // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudWithNormals->points[i].normal_x = normals(i).x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudWithNormals->points[i].normal_y = normals(i).y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudWithNormals->points[i].normal_z = normals(i).z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
+        return pointCloudWithNormals;
+    }
+
+    void visualizePointCloudAndNormalsPCL(
+        const boost::shared_ptr<pcl::PointCloud<pcl::PointXYZRGBNormal>> &pointCloudWithNormals)
+    {
+        auto viewer = boost::make_shared<pcl::visualization::PCLVisualizer>("Viewer");
+
+        int viewRgb(0);
+        viewer->createViewPort(0.0, 0.0, 0.5, 1.0, viewRgb);
+        viewer->addText("Cloud RGB", 0, 0, "RGBText", viewRgb);
+        viewer->addPointCloud<pcl::PointXYZRGBNormal>(pointCloudWithNormals, "cloud", viewRgb); // NOLINT
```

Comment:
It's not needed to have to viewports here

Reply:
My intention was to display the point cloud with color on the left side and the point cloud with color and normals on the right side. Are you saying I should only keep one point cloud with color and normals?

Reply:
I think it looks better with both:

![image](https://user-images.githubusercontent.com/46567886/140270993-b2ab998c-b542-467c-9d63-5fa4197b8728.png)

If there is no color in the point cloud, the right side seems like we are only showing normals.

![image](https://user-images.githubusercontent.com/46567886/140271063-ba8db9db-bdd6-4fec-986b-018dfb7d26b6.png)

Reply:
I don't think we need both, we already have that in another sample

Reply:
That's the point of the CaptureHDRVisNormals sample we have already